packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -807,18 +888,32 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
               throw std::invalid_argument("Promise fn arg count must be 2");
             }
 
+            bool traceTurboModulePromiseRejectionEnabled = getFeatureFlagBoolValue("traceTurboModulePromiseRejectionEnabled");
+            std::shared_ptr<std::string> jsStack;
```

Comment:
Why is this a shared_ptr? You could use `std::optional<std::string>` or just std::string (with len 0) and std::move it to where it's required.

Reply:
nit: name this something like `jsInvocationStack` and/or add a comment to explain we're capturing a stack trace here to provide better error context later.